File: 3702-Longest-Subsequence-With-Non-Zero-Bitwise-XOR/Solution.ts
function longestSubsequence(nums: number[]): number {
    const n = nums.length;
    let xor = 0;
    let hasNonZero = false;

    for (const num of nums) {
        xor ^= num;
        if (num !== 0) {
            hasNonZero = true;
        }
    }

    if (xor !== 0) {
        return n;
    }
    return hasNonZero ? n - 1 : 0;
}

File: 3702-Longest-Subsequence-With-Non-Zero-Bitwise-XOR/Solution.rb
# @param {Integer[]} nums
# @return {Integer}
def longest_subsequence(nums)
    n = nums.length
    xor = 0
    has_non_zero = false

    nums.each do |num|
        xor ^= num
        has_non_zero = true if num != 0
    end

    return n if xor != 0
    has_non_zero ? n - 1 : 0
end
